### PR TITLE
Feature/bare status

### PR DESCRIPTION
**Pull request by jamesmallen**
Addresses #113

### CODE REVIEW

**Comment by mwarkentin** [comment]
Thanks for the submission, I’ll try to take a look soon. 😊

**Comment by mwarkentin** [comment]
/remind me to take a look at this tomorrow.

**Comment by jamesmallen** [comment]
@mwarkentin Did you get a chance to look at this yet?

**Comment by mwarkentin** [comment]
Hey @jamesmallen - initial review looks good, just trying to review the diff in `views.py`.

File: README.rst [hunk 1]
```diff
@@ -172,6 +172,17 @@ requests, you can call ping:
 It will return the text ``pong`` with a 200 status code. Calling this doesn't
 run any of the checks.
 
+Bare status view
```
Comment (mwarkentin) [inline README.rst]: Can you add a `0.15.0 (Unreleased)` heading to the history file and add a bullet for this feature?

File: tests/test_views.py [hunk 2]
```diff
@@ -355,6 +355,37 @@ def test_returns_pong(self):
...
+        reload_settings()
+
+    def test_bare_status_success(self):
+        request = RequestFactory().get('/')
+        response = views.bare_status(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content.decode(), '')
+
+    @patch('watchman.checks._check_databases')
+    @override_settings(WATCHMAN_ERROR_CODE=503)
```
Comment (mwarkentin) [inline tests/test_views.py]: Can you add another test that the default error code for the bare status view will be `500`? You can just copy this one without the override.

File: watchman/views.py [hunk 3]
```diff
@@ -55,109 +52,102 @@ def status(request):
...
+def status(request):
+    checks, ok = run_checks(request)
 
-def ping(request):
-    _deprecation_warnings()
+    if not checks:
+        raise Http404(_('No checks found'))
 
-    return HttpResponse('pong', content_type='text/plain')
+    return checks, 200 if ok else settings.WATCHMAN_ERROR_CODE, {WATCHMAN_VERSION_HEADER: __version__}
```
Comment (mwarkentin) [inline watchman/views.py]: Would you mind pulling the `http_code` out into it's own variable directly above? Just makes this line a bit easier to parse I think (same for in `bare_status`).

File: watchman/views.py [hunk 4]
```diff
@@ -55,109 +52,102 @@ def status(request):
...
 
 
-@auth
 @non_atomic_requests
-def dashboard(request):
-    _deprecation_warnings()
+def bare_status(request):
+    checks, ok = run_checks(request)
```
[the remark this answers is not in the document]
Comment (jamesmallen) [reply watchman/views.py]: For our needs, a 200 even if nothing has been checked is fine - that is a legitimate situation where the webserver is up and responding from our POV (and there might not be any other subsystems to check)

File: README.rst (hunk 1 above)
Comment (jamesmallen) [reply README.rst]: Fixed in 657f135

File: tests/test_views.py (hunk 2 above)
Comment (jamesmallen) [reply tests/test_views.py]: Fixed in 657f135

File: watchman/views.py (hunk 3 above)
Comment (jamesmallen) [reply watchman/views.py]: Fixed in 657f135

File: watchman/views.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mwarkentin) [reply watchman/views.py]: :+1:

File: watchman/views.py (hunk 3 above)
Comment (mwarkentin) [reply watchman/views.py]: @jamesmallen did this get pushed to this branch? I don't see the update on the PR.

**Comment by jamesmallen** [comment]
I accidentally pushed to a different branch. Everything should be fixed (in the correct branch!) in 446c889

**Comment by mwarkentin** [comment]
@jamesmallen Sorry, one more update if possible - could you update the sample project with the required configuration to make this endpoint available at `http://127.0.0.1:8000/watchman/bare/` or similar? And then add a [corresponding example in the readme](https://github.com/mwarkentin/django-watchman/#instructions)?

**Comment by mwarkentin** [comment]
If you’re not sure how to do this, let me know and I can take care of it.

**Comment by jamesmallen** [comment]
Sure thing. Added it in ae48537

**Pull request merged**